graphql-dxm-provider/src/main/java/org/jahia/modules/graphql/provider/dxm/node/GqlJcrNodeImpl.java

```diff
@@ -551,7 +551,15 @@ public String getUrl() {
     @Override
     @GraphQLDescription("Get node thumbnail URL")
     public String getThumbnailUrl(@GraphQLName("name") @GraphQLDescription("Thumbnail name") String name) {
-        return node.getThumbnailUrl(name);
+        try {
+            if (node.hasNode(name)) {
+                return node.getThumbnailUrl(name);
+            }
+        } catch (RepositoryException e) {
+            throw new DataFetchingException(e);
+        }
+
+        return null;
```

Comment:
I am not really sure about the check. It's not because a node have a child node named: toto, that toto is actually a thumbnail.

There is functions available on JCRNodeWrapper to get the list of available thumbnails already:
`public List<String> getThumbnails() {`
That is doing additional checks to not return some node types.

Also I would be in favor of updating the description to reflect that in case thumbnail does not exists it will return null.